tests/console/zypper_lifecycle.pm

```diff
@@ -39,11 +39,17 @@ our $date_re = qr/[0-9]{4}-[0-9]{2}-[0-9]{2}/;
 
 sub run {
     diag('fate#320597: Introduce \'zypper lifecycle\' to provide information about life cycle of individual products and packages');
+
+    # Workaround for bsc#1166549, zypper lifecycle error with 'Error building the cache'.
+    # Currently the only workaround option is to run zypper-lifecycle as root.
+
+    record_soft_failure 'bsc#1166549 - zypper lifecycle failed to create metadata cache directory';
```

Comment:
You record a soft failure but do not check for the bug being present. Please conduct the "happy path" first, i.e. the normal expected flow and detect the error condition. Only then work around recording a soft failure. This way you prevent false positive records for the bug which is a common problem we want to avoid.

Reply:
when the bug presented it hang there with 'zypper lifecycle' till timedout. but in the console it shows permission denied for create metadata dir. like this 
https://openqa.nue.suse.com/tests/4817306#step/zypper_lifecycle/5, any suggestion for the fix?

Reply:
Yes, you can call the command in bash with "timeout", e.g. `timeout 180 zypper lifecycle` and call that in script_run and check the return code.

Reply:
but that'll make the code complicate and hard to understand. the issue now is that when 'zypper lifecycle' hang at the start,all the following 'zypper lifecycle' in this script will also hangs with permission. so  in that case the code will be horrible.

Reply:
and this kind of work should be done by the tool group. which will save us lots of code to do this kind of re-code.

Reply:
I don't see the relation to the tools group. We commonly do not work on os-autoinst-distri-opensuse at all. I don't understand what you think is complicated about the timeout command. You could certainly also use `my $ret = script_run 'zypper lifecycle', 180;` but I do not think it makes a significant difference

Reply:
I still see this unconditional reporting as a problem and I do not plan to approve this PR if we record a soft failure in all the cases where this code is running. Imagine for example QAM jobs. Why should we record a soft failure for older, unaffected SLE versions which do not have this bug? And what if the bug is fixed? There would be likely many more annoying reminder comments in a bug that would not even present anymore.

Reply:
so, how about we add condition like this?
if ((arch == s390x || arch == ppc64le) && is_sle(15-sp3) && is_migration) {
    record_soft_fail;
}

or we just simply remove this record_softfail.

how do you think about this? thanks

Reply:
looks great!

Reply:
thanks for the review.